cl/scrapers/management/commands/clone_from_cl.py

```diff
@@ -0,0 +1,406 @@
+
+            for cite_data in citation_data:
+                # Create citations
+                cite_data["cluster_id"] = opinion_cluster.pk
+                Citation.objects.create(**cite_data)
+
+            for opinion_data in prepared_opinion_data:
+                # Update cluster_id in opinion's json
+                opinion_data["cluster_id"] = opinion_cluster.pk
+
+                # Create opinion
+                op = Opinion.objects.create(**opinion_data)
+
+                # Store created opinion id
+                added_opinions_ids.append(op.id)
+
+            opinion_clusters.append(opinion_cluster)
+            print(
+                ">> View cloned case here:",
+                reverse("view_case", args=[opinion_cluster.pk, docket.slug]),
+            )
+
+        # Add opinions to search engine
+        add_items_to_solr.delay(added_opinions_ids, "search.Opinion")
+
+    return opinion_clusters
+
+
+def clone_docket(
+    session: Session, docket_ids: list, object_type="search.Docket"
+):
+    """
+    Download docket data from courtlistener.com and add it to local
+    environment
+    :param session: a Requests session
+    :param docket_ids: a list of docket ids
+    :param object_type: Docket app name with model name
+    :return: list of docket objects
+    """
+
+    dockets = []
+
+    for docket_id in docket_ids:
+        print(f">> Cloning docket id: {docket_id}")
+
+        model = apps.get_model(object_type)
+
+        try:
+            docket = model.objects.get(pk=docket_id)
+            print(
+                ">> Docket already exists here:",
+                reverse("view_docket", args=[docket.pk, docket.slug]),
+            )
+            dockets.append(docket)
+            continue
+        except model.DoesNotExist:
+            pass
+
+        # Create new Docket
+        docket_endpoint = f"{dockets_endpoint}{docket_id}/"
+        docket_data = session.get(docket_endpoint, timeout=60).json()
+
+        # Remove unneeded fields
+        for f in [
+            "resource_uri",
+            "original_court_info",
+            "absolute_url",
+            "clusters",
+            "audio_files",
+            "tags",
+            "panel",
+        ]:
+            del docket_data[f]
+
+        with transaction.atomic():
+
+            # Get or create required objects
+            docket_data["court"] = (
+                clone_court(session, [docket_data["court"].split("/")[7]])[0]
+                if docket_data["court"]
+                else None
+            )
+
+            docket_data["appeal_from"] = (
+                clone_court(
+                    session, [docket_data["appeal_from"].split("/")[7]]
+                )[0]
+                if docket_data["appeal_from"]
+                else None
+            )
+
+            docket_data["assigned_to"] = (
+                clone_person(
+                    session, [docket_data["assigned_to"].split("/")[7]]
```

Comment:
A helper function for this would probably be good:

```
.split("/")[7]]
```

Perhaps:

```
def get_id_from_url(api_url: str) -> str:
    """Get the PK from an API url"""
    return api_url.split("/")[-1]]
```